src/controllers/krate/versions.rs

```diff
@@ -302,4 +354,228 @@ struct PaginatedVersionsAndPublishers {
 struct ResponseMeta {
     total: i64,
     next_page: Option<String>,
+    release_tracks: Option<ReleaseTracks>,
+}
+
+#[derive(Debug, Eq, PartialEq, Serialize)]
+struct ReleaseTracks(IndexMap<ReleaseSlug, HighestSemver>);
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+enum ReleaseSlug {
+    MajorX(u64),
+    MajorMinor(u64, u64),
+    Major(u64),
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize)]
+struct HighestSemver {
+    highest: SemverTriple,
```

Comment:
why do we not use `semver::Version` here?

Reply:
Since we only need the `major`, `minor`, and `patch`, we don't need to keep the full `semver::Version`.

Reply:
Hmm, after reconsidering, I think keeping `semver::Version` could be useful if we need a full version string. This is likely what we want, since we don't give id here, and if we need to map it back at frontend, we might require a full version string to do so. Therefore, I'll make the change!

Reply:
yeah, exactly